common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -236,41 +237,93 @@ public void recycle(Object object) {
 
         // Let Link extend AtomicInteger for intrinsics. The Link itself will be used as writerIndex.
         @SuppressWarnings("serial")
-        private static final class Link extends AtomicInteger {
+        static final class Link extends AtomicInteger {
             private final DefaultHandle<?>[] elements = new DefaultHandle[LINK_CAPACITY];
 
             private int readIndex;
             private Link next;
```

Comment:
these member variables are accessed outside the class ... should they be package private?

Reply:
yes